src/VisualStudio/Xaml/Impl/Implementation/XamlOleCommandTarget.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    /// <summary>
+    /// This command target routes commands in .xaml files.
+    /// </summary>
+    internal sealed class XamlOleCommandTarget : AbstractOleCommandTarget
```

Comment:
We have a view filter in the XAML Language Service in VS that gets the final say in what shows up. There are no unwanted commands from what I can tell. But even if we do find some we can turn them off there.
I agree though, eventually languages should be able to declare what commands they do or do not support.